### PR TITLE
Restructure getEndpointIds function to pass ui.test.js

### PR DESCRIPTION
**Pull request by ethanzhouyc**
- call API directly without action
- check if $serverGet is defined before calling it
- the change would help #1482 pass unit tests and prevent `ui.test.js` from failing in the future

### CODE REVIEW

File: src/components/ZclCreateModifyEndpoint.vue [hunk 1]
```diff
@@ -544,9 +544,11 @@ export default {
       }
     },
     getEndpointIds() {
-      this.$store.dispatch('zap/getEndpointIds').then((res) => {
-        this.endpointIds = res.data
-      })
+      if (this.$serverGet != null) {
```
Comment (brdandu) [inline src/components/ZclCreateModifyEndpoint.vue:547]: When is `this.$serverGet ` null? What is the reason for this change?

Comment (ethanzhouyc) [reply src/components/ZclCreateModifyEndpoint.vue:547]: `ui.test.js` call `mount` and `shallowMount` functions to simulate creating frontend components and in this case `$serverGet` is null. This check is very common in frontend files

**Review by paulr34 — APPROVED** [review]

**Review by tecimovic — APPROVED** [review]
